README.adoc

```diff
@@ -1,41 +1,116 @@
 == Quick development start
 
-To build and run from source:
+The project is ready to start the application in
+https://tbroyer.github.io/gwt-maven-plugin/codeserver-mojo.html[GWT SuperDevMode].
+In SuperDevMode, the client side of the application is served by GWT's CodeServer, which can quickly
+recompile changes made in the source files in your IDE and serve the updated client upon browser refresh.
+On the other hand, the webapp is deployed to an AppServer running in a separate process and only provides
+the server side of the application.
+This way it is possible to reduce the refresh time during development either by just refreshing browser window
+to get client code changes or by just redeploying the webapp (without recompiling the client) to see backend changes.
 
-----
-$ mvn clean install -DskipTests
+Additionally, the GWT Maven plugin is configured to produce source maps which allow to browse the Java source files,
+set breakpoints and inspect variables using the Developer tools in Chrome.
+
+=== Running SuperDevMode from command line
+
+==== 1. Build the whole project
+
+```bash
+$ mvn clean install -Dgwt.skipCompilation
+```
+
+We can skip GWT compilation because it will be done anyway by the CodeServer
+when accessing the application from browser for the first time.
+
+==== 2. Launch CodeServer
+
+```bash
+$ mvn gwt:codeserver -pl optaplanner-wb-ui -am
+```
+
+The process will keep running in the foreground and thus will occupy the terminal window.
+
+==== 3. Start backend
+
+It's recommended to do this in a different terminal window.
+
+```bash
 $ cd optaplanner-wb-webapp
-$ mvn clean gwt:run
-----
+$ mvn wildfly:start wildfly:deploy-only
```

Comment:
Just idea, we not need to apply. Maybe we could split this step, because if there occurs some problem during execution of `mvn wildfly:start wildfly:deploy-only` and then the same command is reruned there is error about already running application server, so from my point of view we could split this command or extend to kill the server like: `mvn wildfly:start wildfly:deploy-only; mvn wildfly:shutdown`

Reply:
@jomarko In the situation you're describing, you're probably looking for `wildfly:redeploy-only`, which allows you to redeploy the application without having to shutdown the server and start it again.

`mvn wildfly:start wildfly:deploy-only; mvn wildfly:shutdown` doesn't make much sense to me. It will only tell you if the application deploys successfully but you won't be able to work with it as the server will shut down immediately after the application deploys.

Note that, unlike the `gwt:codeserver` goal, most of WildFly goals do not block.

Reply:
@yurloc ok, just wanted to bring out the scenario when some step fail, after your explanation, I am probably fine to keep the steps without change

Reply:
@jomarko Yes. If I'm not misunderstood, you're looking for the best approach when you deploy the application for the first time with `mvn wildfly:start wildfly:deploy-only` but there's a deployment error. You fix it and want to try again. In this scenario you can't just run the previous command again because it's left the WildFly process running in the background (that's expected). Now you have two options:

1. **Kill WildFly and start from scratch.** Run`mvn wildfly:shutdown; mvn wildfly:start wildfly:deploy-only` as one command (or separately). That's what you should be trying after you've read the readme. If you haven't figured this out then the readme is lacking some details (I'll happily add them unless you provide your own suggestions).
This is the basic approach because it's based on the minimal number of goals you need to get the work done but it's not optimal because it involves an unnecessary server startup time overhead.

2. **Just redeploy.** `mvn wildfly:redeploy-only`. I haven't tried this one but it should just redeploy the application without restarting the server. I didn't mention it in the readme because, first, I haven't used it myself and, second, it's an extra piece of information and I wanted to keep the readme brief, but if it works and saves some time I think we should add it there.

So let me know if the second option works for you and if it's what you were looking for and I'll update the readme or feel free to submit your own suggestions via a PR.